File: packages/sdk-csharp/OpenInherit.Sdk/ValidationResult.cs
namespace OpenInherit.Sdk;

/// <summary>
/// Conformance level constants for INHERIT validation.
/// </summary>
public static class ConformanceLevel
{
    /// <summary>Level 1 (JSON Schema) validation failed.</summary>
    public const int Failed = 0;
    /// <summary>The document passed JSON Schema validation.</summary>
    public const int Level1 = 1;
    /// <summary>The document passed both schema and referential integrity checks.</summary>
    public const int Level2 = 2;
}

/// <summary>
/// The outcome of validating an INHERIT document against the bundled schema.
/// </summary>
public record ValidationResult(
    bool Valid,
    string SchemaMode,
    int ConformanceLevel,
    IReadOnlyList<ValidationError> Errors,
    string Disclaimer
);

/// <summary>
/// A single validation error, with a JSON Pointer path and a human-readable message.
/// </summary>
public record ValidationError(string Path, string Message, int Level = 1);

File: packages/sdk-java/src/main/java/org/openinherit/sdk/ValidationResult.java
package org.openinherit.sdk;

import java.util.List;

public record ValidationResult(
    boolean valid,
    String schemaMode,
    List<ValidationError> errors,
    String disclaimer,
    int conformanceLevel
) {
    /**
     * Represents a single validation error.
     *
     * @param path  JSON path to the failing element
     * @param message human-readable description of the failure
     * @param level  1 = schema error, 2 = referential integrity error
     */
    public record ValidationError(String path, String message, int level) {
        /** Convenience constructor for Level 1 (schema) errors. */
        public ValidationError(String path, String message) {
            this(path, message, 1);
        }
    }

    public static final String DISCLAIMER =
        "Validation results are informational. They verify schema " +
        "conformance and data structure, not legal accuracy or " +
        "completeness. Do not rely on validation results as the " +
        "sole basis for legal or financial decisions.";
}
